lib/src/publication/remote.dart

```diff
@@ -97,6 +98,10 @@ class RemoteTrackPublication<T extends RemoteTrack>
   // from one of the renderers
   void _onVideoRendererVisibilityUpdateEvent(
       TrackVisibilityUpdatedEvent event) {
+    if (event.track is LocalVideoTrack) {
```

Comment:
question: why is the RemoteTrackPublication handling events from a LocalVideoTrack?

Reply:
Right! this should never happen so the check is not necessary. I will remove it.